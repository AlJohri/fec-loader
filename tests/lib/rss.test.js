const { ObjectReadableMock, ObjectWritableMock } = require('stream-mock');

jest.mock('cross-fetch');
const fetch = require('cross-fetch');

const originalProcess = process;

global.process = {
    ...originalProcess,
    mainModule: {
        filename: 'rss.test.js'
    }
};

const rss = require('../../lib/rss');

global.process = originalProcess;

test('rss', async done => {
    const input = `<?xml version="1.0" encoding="UTF-8"?>
        <rss xmlns:dc="http://purl.org/dc/elements/1.1/" version="2.0">
          <channel>
            <item>
              <title>New filing by SALTCHUK RESOURCES, INC. PAC</title>
              <link>http://docquery.fec.gov/dcdev/posted/1354911.fec</link>
              <description>&lt;p&gt;The SALTCHUK RESOURCES, INC. PAC successfully filed  their F3XN OCTOBER MONTHLY with the coverage period of 09/01/2019 to 09/30/2019 and a confirmation ID of FEC-1354911&lt;/p&gt;*********CommitteeId: C00411694 | FilingId: 1354911 | FormType: F3XN | CoverageFrom: 09/01/2019 | CoverageThrough: 09/30/2019 | ReportType: OCTOBER MONTHLY*********</description>
              <pubDate>Sat, 05 Oct 2019 19:03:06 GMT</pubDate>
              <guid>http://docquery.fec.gov/dcdev/posted/1354911.fec</guid>
              <dc:date>2019-10-05T19:03:06Z</dc:date>
            </item>
          </channel>
        </rss>`;

    const reader = new ObjectReadableMock(input);
    const writer = new ObjectWritableMock();

    fetch.mockReturnValue(
        Promise.resolve({
            body: reader
        })
    );

    const stream = await rss({});

    stream.pipe(writer);

    writer.on('finish', () => {
        expect(fetch.mock.calls).toMatchSnapshot();
        expect(writer.data).toMatchSnapshot();
        done();
    });
    
});
