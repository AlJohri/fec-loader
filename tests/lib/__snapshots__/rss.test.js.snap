// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rss 1`] = `
Array [
  Array [
    "http://efilingapps.fec.gov/rss/generate?preDefinedFilingType=ALL",
  ],
]
`;

exports[`rss 2`] = `
Array [
  Object {
    "committee_id": "C00411694",
    "committee_name": "SALTCHUK RESOURCES, INC. PAC",
    "coverage_end_date": "09-30-2019",
    "coverage_start_date": "09-01-2019",
    "document_description": "OCTOBER MONTHLY 2019",
    "fec_file_id": "1354911",
    "fec_url": "http://docquery.fec.gov/dcdev/posted/1354911.fec",
    "form_type": "F3XN",
  },
]
`;
